README.md

```diff
@@ -221,16 +221,86 @@ root
+```
+
+Running `go build ./...` to build all packages at the root level will build all packages, but it will discard the 
+resulting artifacts; it therefore serves only as a check that the packages _can_ be built. Note that building the 
+enclave using `go` will compile it for a non-SGX mode and allow it to be run for test purposes. Compiling for SGX mode 
+requires `ego-go ` from [Ego](https://www.edgeless.systems/products/ego/) to be used in placement.  
+
+
+### Running the tests
+The tests require an Obscuro enclave to be locally running, and as such the image should first be created and added to the 
+docker images repository. Building the image is described in [dockerfiles](dockerfiles) and can be performed using the 
+below in the root of the project;
+
+```
+docker build -t obscuro_enclave -f ./dockerfiles/enclave.Dockerfile .
+```
+
+To run all unit, integration and simulation tests locally, run the below in the root of the project;
+
+```
+go test ./...
+```
+
+### Starting a local testnet
+At the moment running a local testnet has an additional dependency on [jq](https://stedolan.github.io/jq/) and the module 
+should be additionally installed prior to starting the local testnet. See the `jq` documentation for installation
+instructions. 
+
+The testnet is constructed from docker images that have all executables built, installed and available for running. The 
+images are created from the working directory of the repository checkout. To build the images and to add them into the 
+docker images repository use;
+
+```
+cd ./testnet && ./testnet-local-build_images.sh 
+```
+
+The above will perform all the relevant builds and ensure the images are ready for running each component, and as such 
+it takes in the order of 4-5 mins to complete. The following images are created;
+
+```
+testnetobscuronet.azurecr.io/obscuronet/obscuro_enclave            # the enclave 
+testnetobscuronet.azurecr.io/obscuronet/obscuro_gethnetwork        # the L1 network 
+testnetobscuronet.azurecr.io/obscuronet/obscuro_host               # the host
+testnetobscuronet.azurecr.io/obscuronet/obscuro_contractdeployer   # deploys the management contract to the host
+```
+
+To start the test network locally run the below scripts. Note that it is recommended to use the scripts with arguments 
+as detailed below. The arguments are set to correspond to valid pre-determined public / private key pair values for 
+contract deployment and roll up publishing (where `pkstring` is the private key pair for the first `pkaddress` supplied 
+in `testnet-local-gethnetwork.sh`). Using these values, and starting with a nonce of zero, means the addresses of the 
+contracts deployed are known a-priori, and so can be supplied in the `start-obscuro-node.sh` script as shown. As only 
+a single Obscuro node is started, it must be set as a genesis node. 
+
+```
+./testnet-local-gethnetwork.sh --pkaddresses=0x13E23Ca74DE0206C56ebaE8D51b5622EFF1E9944,0x0654D8B60033144D567f25bF41baC1FB0D60F23B
```

Comment:
I'd add a bit more details here as it still feels like magic. @otherview, pls correct me if I'm wrong:
- 0x13E23Ca74DE0206C56ebaE8D51b5622EFF1E9944:  Address to pre-fund, from which  the obscuro management contract will be deployed. Has to be known by the obscuro nodes
- x0654D8B60033144D567f25bF41baC1FB0D60F23B:  Address to pre-fund, from which  an erc20  contract will be deployed. Has to be known by the obscuro nodes

Reply:
- 0x13E23Ca74DE0206C56ebaE8D51b5622EFF1E9944 : It's the Worker wallet, it's used to deploy the Obscuro Management Contract + the ERC20 (+ will be used for the transaction injector in the future)

- 0x0654D8B60033144D567f25bF41baC1FB0D60F23B : It's the Obscuro Node 01 Wallet, used to pay for Node 01 rollup transaction costs.

These are different addresses to avoid Nonce conflicts.
If there were more nodes, we would likely need to add more addresses.

Hope that makes sense ☺️

Reply:
Done